From 171b14b660f35f593748ac62bbdbb43ace2c582d Mon Sep 17 00:00:00 2001
From: John Crispin <john@phrozen.org>
Date: Tue, 1 Nov 2016 01:44:15 +0100
Subject: [PATCH 10/22] net: dsa: make the slave device inheret the MAC of the
 parent

This patch makes all slave devices inherit the parent devices MAC.

Signed-off-by: John Crispin <john@phrozen.org>
---
 net/dsa/slave.c |   11 +++++++++++
 1 file changed, 11 insertions(+)

Index: linux-4.9.34/net/dsa/slave.c
===================================================================
--- linux-4.9.34.orig/net/dsa/slave.c
+++ linux-4.9.34/net/dsa/slave.c
@@ -175,6 +175,18 @@ static int dsa_slave_close(struct net_de
 	return 0;
 }
 
+static int dsa_slave_init(struct net_device *dev)
+{
+	struct dsa_slave_priv *p = netdev_priv(dev);
+	struct net_device *master = p->master;
+	struct sockaddr sa;
+
+	ether_addr_copy(sa.sa_data, master->dev_addr);
+	eth_mac_addr(dev, &sa);
+
+	return 0;
+}
+
 static void dsa_slave_change_rx_flags(struct net_device *dev, int change)
 {
 	struct dsa_slave_priv *p = netdev_priv(dev);
@@ -1012,6 +1024,7 @@ static const struct ethtool_ops dsa_slav
 static const struct net_device_ops dsa_slave_netdev_ops = {
 	.ndo_open	 	= dsa_slave_open,
 	.ndo_stop		= dsa_slave_close,
+	.ndo_init		= dsa_slave_init,
 	.ndo_start_xmit		= dsa_slave_xmit,
 	.ndo_change_rx_flags	= dsa_slave_change_rx_flags,
 	.ndo_set_rx_mode	= dsa_slave_set_rx_mode,
